frontend/src/components/HobbyDetail.tsx

```diff
@@ -0,0 +1,74 @@
+import {useEffect, useState} from "react";
+import axios from "axios";
+import {useLocation, useNavigate, useParams} from "react-router-dom";
+import ActivityItem from "./ActivityItem";
+import {Activity, Hobby} from "../models.ts";
+import {Button, Grid} from "@mui/material";
+import styled from "@emotion/styled";
+
+export default function HobbyDetail() {
+    const [hobby, setHobby] = useState<Hobby | null>(null);
+    const [activities, setActivities] = useState<Activity[] | null>(null);
+    const params = useParams();
+    const navigate = useNavigate();
+
+    const location = useLocation();
+    const selectedColor = location.state?.selectedColor || "#f2f2f2";
+
+    useEffect(() => {
```

Comment:
maybe put that in your custom hook 🤙🏻

Reply:
Thanks for the great conversation before - now that makes more sense to me! 👍🏻